src/NetMQ/Core/Patterns/XPub.cs

```diff
@@ -176,15 +184,29 @@ protected override void XReadActivated(Pipe pipe)
                         // If the subscription is not a duplicate, store it so that it can be
                         // passed to used on next recv call.
                         if (m_options.SocketType == ZmqSocketType.Xpub && (unique || m_verbose))
-                            m_pending.Enqueue(sub.CloneData());
+                        {
+                            m_pending.Enqueue(sub);
+                        }
+                        else
+                        {
+                            sub.Close();
+                        }
+
                     }
                 }
+                else if (m_broadcastEnabled && size > 0 && sub.HasMore && sub[0] == 2)
```

Comment:
So that is not the correct way to check if this is the first frame. What if you have 3 frames? what if it is one frame?

So current NetMQ implementation doesn't support multi-messages from xsub to xpub (ZeroMQ is actually do supporting). Anyway I suggest we keep at the beginning for 1 frame and later add support to more frames.